src/main/java/me/drepic/proton/Proton.java

```diff
@@ -43,20 +43,24 @@ public void onEnable() {
 
         boolean useRabbitMQ = config.getBoolean("rabbitMQ.useRabbitMQ");
         boolean useRedis = config.getBoolean("redis.useRedis");
+
+        if (!useRabbitMQ && !useRedis) {
+            logger.log(Level.SEVERE, "Neither RabbitMQ nor Redis is enabled. Shutting down.");
+            getServer().getPluginManager().disablePlugin(this);
+            return;
+        }
+
         if (useRabbitMQ) {
             try {
                 setupRabbitMQ(clientName, groups);
             } catch (IOException | TimeoutException e) {
                 e.printStackTrace();
+                logger.severe("Failed to setup RabbitMQ Connection");
                 getServer().getPluginManager().disablePlugin(this);
                 return;
             }
-        } else if (useRedis) {
-            setupRedis(clientName, groups);
         } else {
-            logger.log(Level.SEVERE, "Neither RabbitMQ nor Redis is enabled. Shutting down.");
-            getServer().getPluginManager().disablePlugin(this);
-            return;
+            setupRedis(clientName, groups);
```

Comment:
can setupRedis throw exceptions like setupRabbitMQ? If so, probably need to wrap in the same try-catch as above

Reply:
The exceptions for rabbitMQ are from the connect method. The redis connect method doesn't have anything that would throw an exception.